client/src/components/Profile.jsx

```diff
@@ -0,0 +1,150 @@
+import React, { Component } from 'react';
+import { Row, Col, Input, Button } from 'react-materialize';
+import { server_url } from '../config.json';
+
+
+export default class Profile extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            Edit: false,
+            msg: '',
+            user: [],
+        };
+
+        this.UpdateClick = this.UpdateClick.bind(this);
+        this.onChangeHandler = this.onChangeHandler.bind(this);
+        this.handleClick_Main = this.handleClick_Main.bind(this);
+        this.handleClick_Edit = this.handleClick_Edit.bind(this);
+    }
+
+    componentDidMount() {
+        fetch(server_url + '/profile/' + window.location.href.split("/").pop())
+        .then(response => response.json())
+        .then(json => {
+            console.log(json);
+            this.setState({ user: json });
+        })
+    }
+
+    componentWillUnmount() {
+        clearInterval(this.timer);
+    }
+
+    UpdateClick(eve) {
+        eve.preventDefault()
+        if (this.state.new_password !== null) {
+          if (this.state.new_password === this.state.repeat_password) {
+            const data = {
+                "email": this.state.email,
+                "first_name": this.state.f_name,
+                "last_name": this.state.l_name,
+                "nickname": this.state.nickname,
+                "location": this.state.location,
+                "password": this.state.new_password,
+                "id": this.state.user.id
+            }
+            console.log(data)
+            fetch(server_url + '/profile/' + window.location.href.split("/").pop(), {
+                method: 'PUT',
+                headers: {'Content-Type':'application/json',
+                'Access-Control-Allow-Origin':'http://localhost:3000'},
+                body: JSON.stringify(data)
+            })
+            window.Materialize.toast("Profile Updated", 2500);
+          }
+          else {
+            window.Materialize.toast("Incorrect repeated password!", 2500);
+
+          }
+        }
+
+    }
+
+    onChangeHandler = (e) => {
+        let { id } = e.currentTarget
+        this.setState({ [id]: e.currentTarget.value })
+    }
+
+    handleClick_Main() {
+        window.location.reload();
```

Comment:
when you change the `state`, the component instantly re-render. You don't need to reload whole page.